my_vulkan/device_memory.hpp

```diff
@@ -60,9 +61,11 @@ namespace my_vulkan
         void set_data(const std::vector<T>& data);
         void set_data(const void* data, size_t size);
         VkDeviceMemory get();
+        [[nodiscard]] std::optional<int> get_external_handle(VkExternalMemoryHandleTypeFlagBits externalHandleType) const;
```

Comment:
would it leak otherwise? just sayin', coz if not then in theory we should stick this on all value returning methods, and I think thats too noisy. I think I would restrict it then to those cases where something bad happens otherwise, what do you think?

Reply:
I put the flag there to avoid warnings, I can remove it. But I thought it won't hurt to always gradually add those correct flags?

Reply:
hmm, then constexpr. noexcept? what was the warning?

Reply:
warnings from clang-tidy, basically it will suggest `[[nodiscard]]` for all const functions which return value. I think it makes sense: const function has no other use except returning the value. It is nonsense to use a const function without checking the return value. As you said: in theory we should put it everywhere. In most cases without it will be fine too. But I did not see any side effect with this mark neither. Why should we restrict it?